src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -491,33 +406,26 @@ def load_particles(
         self.m = particle.mass.to(u.kg).value
         self._particle = particle
 
+        if self.q != 0:
+            self._required_quantities += ["E_x", "E_y", "E_z", "B_x", "B_y", "B_z"]
+
         if x.shape[0] != v.shape[0]:
             raise ValueError(
                 "Provided x and v arrays have inconsistent numbers "
                 " of particles "
                 f"({x.shape[0]} and {v.shape[0]} respectively)."
             )
         else:
-            self.nparticles: int = x.shape[0]
+            self.num_particles: int = x.shape[0]
 
         self.x = x.to(u.m).value
         self.v = v.to(u.m / u.s).value
 
-    def _is_quantity_defined_on_one_grid(self, quantity: str) -> bool:
-        r"""
-        Check to ensure the provided quantity string is defined on at least one grid.
-
-        Returns ``True`` if the quantity is defined on at least one grid and
-        ``False`` if none of the grids have defined the specified quantity.
-        """
-
-        return any(quantity in grid.quantities for grid in self.grids)
-
     def _validate_stopping_inputs(
```

Comment:
The input changes to this function don't really change functionality - they just update it to what it was already supposed to do. Before, you had to include a material for every grid. Now, you can only include it for some, and stopping will only be added on those grids.